### PR TITLE
NMA-347 Large/Small Pin Support (upgrade support)

### PR DESCRIPTION
**Pull request by tomasz-ludek**
Handling a variety of cases of wallet upgrade from versions before rebranding.
Added the support for custom PIN lengths (other than 4 digits).

This changeset touches many different parts of the app and should be tested against all the creating/restoring wallets scenarios as well as regular and automatic authentication, changing the PIN, etc.

This is the inseparable part of the new security model.

### CODE REVIEW

**Comment by HashEngineering** [comment]
@tomasz-ludek since the parent story was merged to redesign, I set the branch for this PR to redesign.  There are a few merge conflicts.

**Comment by sambarboza** [comment]
Screen Shot 2020-02-12 at 2 36 28 PM

It's not showing the "OK/Continue" button for me. Nexus S (emulator) - Android 7.1

**Comment by sambarboza** [comment]
Screen Shot 2020-02-12 at 2 38 19 PM
After entering a wrong pin, it won't let me try with a new pin.

**Comment by tomasz-ludek** [comment]
@sambarboza both issues fixed

File: wallet/src/de/schildbach/wallet/ui/CheckPinDialog.kt [hunk 1]
```diff
@@ -67,6 +67,8 @@ open class CheckPinDialog : DialogFragment() {
     protected var fingerprintHelper: FingerprintHelper? = null
     protected lateinit var fingerprintCancellationSignal: CancellationSignal
 
+    private var pinLength = PinPreviewView.DEFAULT_PIN_LENGTH
```
Comment (sambarboza) [inline wallet/src/de/schildbach/wallet/ui/CheckPinDialog.kt]: Why is this var initialized here with `PinPreviewView.DEFAULT_PIN_LENGTH` as value if it's changed to `WalletApplication.getInstance().configuration.pinLength` on the `onCreateDialog` method?

Am I missing something?

File: wallet/src/de/schildbach/wallet/ui/CheckPinDuringUpgradeDialog.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package de.schildbach.wallet.ui
```
Comment (sambarboza) [inline wallet/src/de/schildbach/wallet/ui/CheckPinDuringUpgradeDialog.kt]: Please add copyright & info header.

File: wallet/src/de/schildbach/wallet/ui/CheckPinDialog.kt (hunk 1 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/CheckPinDialog.kt]: Fixed

File: wallet/src/de/schildbach/wallet/ui/CheckPinDuringUpgradeDialog.kt (hunk 2 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/CheckPinDuringUpgradeDialog.kt]: Fixed

**Comment by tomasz-ludek** [comment]
Both issues fixed.

**Comment by HashEngineering** [comment]
This story may be merged manually, we shall see how testing goes.

**Comment by HashEngineering** [comment]
This was manually merged and squashed as in 89b0e1dc65dc786896a8b7f2f8b54586e46c5a58

**Pull request closed without merging**